Source/Core/InputCommon/ControlReference/ExpressionParser.cpp

```diff
@@ -361,29 +375,25 @@ Device::Control* ControlFinder::FindControl(ControlQualifier qualifier) const
     return device->FindOutput(qualifier.control_name);
 }
 
-class Parser
+struct ParseResult
 {
-public:
-  Parser(std::vector<Token> tokens_, ControlFinder& finder_) : tokens(tokens_), finder(finder_)
+  ParseResult(ParseStatus status_, std::unique_ptr<Expression>&& expr_ = {})
+      : status(status_), expr(std::move(expr_))
   {
-    m_it = tokens.begin();
   }
 
-  ParseStatus Parse(Expression** expr_out)
-  {
-    ExpressionNode* node;
-    ParseStatus status = Toplevel(&node);
-    if (status != ParseStatus::Successful)
-      return status;
-
-    *expr_out = new Expression(node);
-    return ParseStatus::Successful;
-  }
+  ParseStatus status;
+  std::unique_ptr<Expression> expr;
+};
 
+class Parser
+{
+public:
+  Parser(std::vector<Token> tokens_) : tokens(tokens_) { m_it = tokens.begin(); }
```

Comment:
This should probably be an explicit constructor. The token vector can also be `std::move`d here.